### DISCUSSION TITLE
Share change in process for agencies building sites with the site editor

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#230] [Topic Ideas]
From: https://github.com/WordPress/developer-blog-content/issues/229

> This has come up repeatedly over the years with the Site Editor: switching to the Site Editor is both a technical and process change. While we have lots of resources on the technical side, can we do a post featuring the process side too? I think having a few examples of "How we do it" style posts would go a long way to help folks consider switching by showing them _how_ folks have done so.

CC: @annezazu, @bbertucc

**Comment by justintadlock** [comment]
@bbertucc - Just pinging you here on the discussion. The topic proposal is a bit open-ended. If you're interested in writing this, it'd be great to get a few more details on what this article would cover (no need to do a full outline or anything, just a general synopsis). This is to ensure that the Editorial Group can decide if the topic should be approved.

We have a regular monthly meeting on the first Thursday of the month at 13:00 UTC, in which the Editorial Group approves topics. The next one will happen on Thursday, March 7.

**Reply by bbertucc** [reply]
Hi @justintadlock. Here is an article I pulled together (FWIW- Google Gemini Expert helped me, but I reviewed and made a lot of changes and hope y'all make changes too). Once things are set, I can grab a few screengrabs and maybe a cat video if you like.

## Decubing's Journey with WordPress Full Site Editing: Benefits, Challenges, and Evolving Design

My web development agency, [Decubing](http://decubing.com), has embraced Full Site Editing (FSE) fully. FSE tools transformed our process and enabled us to deliver better experiences for our clients.

Let's delve into the key reasons we embrace FSE on production sites, as well as a critical limitation that still needs to be addressed.

### FSE Benefits

**Design Democratization**: FSE empowers everyone to become a designer. We now see a much stronger focus on providing design training for both stakeholders and developers.  Expert designers remain essential to tackle complex design challenges and train new team members, but FSE unlocks a broader spectrum of creativity within our teams and client organizations.

**Content-First Approach**: With FSE, we can now start by inputting content immediately after the project kickoff. This allows us to adapt patterns or build new designs based on real content, giving us a more accurate and organic vision of the final product. We also work more closely with clients, scheduling regular review meetings during our iterative design process. 

**Minimizing Custom Code**:  FSE's rich library of blocks and the availability of powerful plugins frequently eliminate the need for extensive custom coding. This frees up time for focusing on the design and user experience, while also minimizing technical debt. 

**Beyond Mockups**: Instead of static mockups, we start with a flexible style guide (colors, fonts, etc.) and a curated selection of inspirational examples.  FSE's global style editor allows us to efficiently translate these design guidelines directly into the working website. We're amazed how happy clients are with just a few changes to font and color.

### FSE Challenges

**Supporting Screen Readers and Keyboards**: A major shortcoming of FSE is its current limitations in providing a screen-reader-friendly experience. Since much of FSE relies on drag-and-drop interactions, certain functions remain difficult to access for a keyboard-reliant screen reader user. This creates a significant barrier to entry. We often find ourselves temporarily switching off the block editor for sites where screen reader accessibility is paramount, but this shouldn't be an either/or situation. 

**Rapidly Changing Tech**: I love the rapid speed of innovation, but clients often complain of issues related to quick changes in the FSE project. One particular problem is around the evolving support of fonts. In the last few years, we've had to spend many hours supporting how fonts work across our site network.

**Reduced Revenue**: Decubing charges by the hour. Since FSE reduced development time, our revenue is reduced. We've been searching for ways to improve revenue. We've seen some increase in the time we bill for training and we now offer FSE-focused hosting services. There is some talk about pivoting our agency into the TikTok star model, where we develop FSE training videos, but I'm holding out as long as I can before videos where I train my cat to use WordPress. Meow. 

### Overall

Despite any challenges, I firmly believe that FSE is a benefit for our agency. We've cut down development time, focused more on design, and given clients greater freedom to iterate. In other words, FSE has allowed us to give clients more creative control. More creative control means clients can create whatever they dream, which is the reason I started Decubing.

**Reply by justintadlock** [reply]
It was undecided whether this topic should be approved in the March Editorial Group meeting.

My personal concern is that the above is not in-depth enough on its own. It seems like this was the full article posted above, and I think that's where some of the confusion is. If this is meant as an outline for a beefier post, I think we can get it approved.

Other feedback from the meeting:

> In my opinion the topic for the agencies is generally interesting, I would love to read a real case study from the respected agency how they adopted to changes. I agree that the post shared in the ticket is too general, it doesn't seem to offer much insight for agencies looking to adopt their processes to the block editor way of building websites

As a side note, we generally avoid linking to third-party sites. I'm not sure how that should work for case studies, though.

**Reply by bbertucc** [reply]
Cool. You can think of it as an outline. Those specific areas can be beefed up with more examples. That said, I'm not sure I have enough time to complete it. Jamming everything into the next three weeks before I go on paternity leave. If someone wanted to ask me questions that I could answer, I would be game to answer those and get ChatGPT to spin it into an article or something. Beyond that, this work might have to wait until after April.

**Reply by justintadlock** [reply]
Waiting until after April would be OK.

**Reply by bbertucc** [reply]
cool. feel free to ping me.

**Reply by colorful-tones** [reply]
> In my opinion the topic for the agencies is generally interesting, I would love to read a real case study from the respected agency how they adopted to changes. I agree that the post shared in the ticket is too general, it doesn't seem to offer much insight for agencies looking to adopt their processes to the block editor way of building websites

I agree with this. I am concerned that we focus on the steps and observations to target our outcomes.

[Anne's initial call-to-action](https://github.com/WordPress/developer-blog-content/issues/229#issue-2157387516) (using to contextualize understanding):

> This has come up repeatedly over the years with the Site Editor: switching to the Site Editor is both a technical and process change. While we have lots of resources on the technical side, can we do a post featuring the process side too? I think having a few examples of "How we do it" style posts would go a long way to help folks consider switching by showing them how folks have done so.

A lot of the mechanics of the training are available (creating a theme, working with patterns, locking things down). I'm not stating that they're conclusive and exhaustive, but just recognizing that some of them already exist. So, the focus on processes, planning, and agency workflows would be a refreshing topic.

> Minimizing Custom Code: FSE's rich library of blocks and the availability of powerful plugins frequently eliminate the need for extensive custom coding. This frees up time for focusing on the design and user experience, while also minimizing technical debt.

I would read that post ☝️ 

> Beyond Mockups: Instead of static mockups, we start with a flexible style guide (colors, fonts, etc.) and a curated selection of inspirational examples. FSE's global style editor allows us to efficiently translate these design guidelines directly into the working website. We're amazed how happy clients are with just a few changes to font and color.

I feel like there is an opportunity to collaborate with and perhaps offer insights on #175 

> Reduced Revenue: Decubing charges by the hour. Since FSE reduced development time, our revenue is reduced. We've been searching for ways to improve revenue. We've seen some increase in the time we bill for training and we now offer FSE-focused hosting services. There is some talk about pivoting our agency into the TikTok star model, where we develop FSE training videos, but I'm holding out as long as I can before videos where I train my cat to use WordPress. Meow.

Many folks (agencies of one or more) would love to hear about your observations. Side note (previous freelancer and agency dev here 😉 ): consider charging more. Your value and understanding of WordPress has excelled if you're learning and growing and that value could potentially be allotted to an hourly increase.

> That said, I'm not sure I have enough time to complete it. Jamming everything into the next three weeks before I go on paternity leave.

Congratulation! I encourage you to enjoy time with your family; open source will be here. ❤️‍🔥 

> Despite any challenges, I firmly believe that FSE is a benefit for our agency. We've cut down development time, focused more on design, and given clients greater freedom to iterate. In other words, FSE has allowed us to give clients more creative control. More creative control means clients can create whatever they dream, which is the reason I started Decubing.

I love this so much! The fact that you're circling back and willing to share your experiences is so impactful and an excellent contribution @bbertucc ❤️

**Comment by colorful-tones** [comment]
For what it's worth - this just came up in the WPACF.Slack.com community. 

> It’s very hard to find anywhere a progress and a talk about FSE version control. I’m not talking about revisions, as this is database dependent - but rather how could we keep the functionality of the project synced via Git.

I have definitely heard this from a few community developers, and I've opened [a separate discussion to track](https://github.com/WordPress/developer-blog-content/discussions/249), but it could possibly be related to this broader "agencies building sites" topic. Thanks!

**Comment by bph** [comment]
This is such a broad topic. From my conversations with agencies, each have found a way to use create block theme to export changes stored into the database and add them to either theme.json, like style variations or create separate files, i.e. templates. The team around that plugin is also working on exporting patterns. 

WebDev Studios created a [block starter theme](https://github.com/WebDevStudios/wds-bt) they shared publicly. You can read more about it on their blog: [Embracing the Future: How WebDevStudios Is Leading with WordPress Block Themes](https://webdevstudios.com/2024/08/26/webdevstudios-wordpress-block-themes/).

An earlier [Hallway Hangout covered the process to use Playground, GitHub and Create block theme plugin](https://make.wordpress.org/core/2024/06/25/recap-hallway-hangout-theme-building-with-playground-create-block-theme-plugin-and-github/) to submit pull requests to theme's GitHub repo.  The recap post has a short demo and the instructions on how to set it up. 

Those are only two different methods. 
Even if we collected more examples, because of our external link policies, we won't be able to share them in an article on the Developer Blog.

**Comment by bph** [comment]
For the purpose of the Developer Blog, I close this discussion. If a writer wants to tackle this, I'd be happy to work with them to get it published on the Gutenberg Times.
